python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -94,103 +123,234 @@ def optimize(self, model,
         :param use_ipex: (optional) if not NOne, then will only find the
                model with this specific ipex setting
-        :param allow_acc: (optional) a float represents the accuracy threshold
-               that can be tollerated.
-
-        :return: best model
+        :param :param accuracy_criterion: (optional) a float represents tolerable
+               accuracy drop percentage, defaults to None meaning no accuracy control.
+        :return: best model, corresponding acceleration option
         '''
-        pass
+        invalidOperationError(len(self.optimized_model_dict) > 0,
+                              "There is no optimized model. You should call .optimize() \
+                              before get_best_model()")
+        invalidInputError(accelerator in [None, 'onnxruntime', 'openvino', 'jit'],
+                          "Only support accelerator 'onnxruntime', 'openvino' and 'jit'.")
+        # TODO: include fp16?
+        invalidInputError(precision in [None, 'int8', 'bf16'],
+                          "Only support precision 'int8', 'bf16'.")
+
+        best_model = self.optimized_model_dict["original"]["model"]
+        best_metric = CompareMetric("original",
+                                    self.optimized_model_dict["original"]["latency"],
+                                    self.optimized_model_dict["original"]["accuracy"])
+
+        for method in self.optimized_model_dict.keys():
+            if method == "original":
+                continue
+            option = ALL_INFERENCE_ACCELERATION_METHOD[method]
+            result = self.optimized_model_dict[method]
+            if accelerator is not None:
+                if not getattr(option, accelerator):
+                    continue
+            if precision is not None:
+                if precision == 'bf16' and not option.bf16:
+                    continue
+                if precision == 'int8' and not option.inc:
+                    continue
+            if use_ipex:
+                if not option.ipex:
+                    continue
+
+            if accuracy_criterion is not None:
+                accuracy = result["accuracy"]
+                compare_acc = best_metric.accuracy
+                if self.direction == "min":
+                    if (accuracy - compare_acc) / compare_acc > accuracy_criterion:
+                        continue
+                else:
+                    if (compare_acc - accuracy) / compare_acc > accuracy_criterion:
+                        continue
+
+            # After the above conditions are met, the latency comparison is performed
+            if result["latency"] < best_metric.latency:
+                best_model = result["model"]
+                best_metric = CompareMetric(method, result["latency"], result["accuracy"])
+
+        return best_model, _format_acceleration_info(best_metric.method_name)
+
+    @staticmethod
+    def trace(model: nn.Module,
```

Comment:
move the implementation of `Trainer.trace` and `Trainer.quantize` to `Optimizer`, and have `Trainer` to call `Optimizer`

Reply:
We may split this one to next PR to make the PR smaller and easier to control. The `quantize` and `trace` API will be migrate to InfereceOptimizer in 4 steps:

0. This PR: InferenceOptimizer call Trainer.
1. PR + 1: Trainer call InferenceOptimizer.
2. PR + 2: Revise our example, use-case.
3. PR + 3: deprecate Trainer's quantize and trace API.